src/cdk-experimental/dialog/dialog-container.ts

```diff
@@ -56,155 +50,115 @@ export function throwDialogContentAlreadyAttachedError() {
+   * after the dialog is closed.
+   */
+  _closeInteractionType: FocusOrigin | null = null;
 
-  /** Stream of animation `done` events. */
-  readonly _animationDone = new Subject<AnimationEvent>();
+  /** ID of the element that should be considered as the dialog's label. */
+  _ariaLabelledBy: string | null;
 
   constructor(
-    private _elementRef: ElementRef<HTMLElement>,
-    private _focusTrapFactory: FocusTrapFactory,
-    private _changeDetectorRef: ChangeDetectorRef,
-    private readonly _interactivityChecker: InteractivityChecker,
-    private readonly _ngZone: NgZone,
+    protected _elementRef: ElementRef,
+    protected _focusTrapFactory: FocusTrapFactory,
     @Optional() @Inject(DOCUMENT) _document: any,
-    /** The dialog configuration. */
-    public _config: DialogConfig,
+    @Inject(DialogConfig) public _config: C,
+    private _interactivityChecker: InteractivityChecker,
+    private _ngZone: NgZone,
+    private _overlayRef: OverlayRef,
+    private _focusMonitor?: FocusMonitor,
   ) {
     super();
-
+    this._ariaLabelledBy = this._config.ariaLabelledBy || null;
     this._document = _document;
-
-    // We use a Subject with a distinctUntilChanged, rather than a callback attached to .done,
-    // because some browsers fire the done event twice and we don't want to emit duplicate events.
-    // See: https://github.com/angular/angular/issues/24084
-    this._animationDone
-      .pipe(
-        distinctUntilChanged((x, y) => {
-          return x.fromState === y.fromState && x.toState === y.toState;
-        }),
-      )
-      .subscribe(event => {
-        // Emit lifecycle events based on animation `done` callback.
-        if (event.toState === 'enter') {
-          this._autoFocus();
-          this._afterEnter.next();
-          this._afterEnter.complete();
-        }
-
-        if (event.fromState === 'enter' && (event.toState === 'void' || event.toState === 'exit')) {
-          this._returnFocusAfterDialog();
-          this._afterExit.next();
-          this._afterExit.complete();
-        }
-      });
   }
 
-  /** Initializes the dialog container with the attached content. */
-  _initializeWithAttachedContent() {
+  ngAfterViewInit() {
+    this._focusTrap = this._focusTrapFactory.create(this._elementRef.nativeElement);
+
     // Save the previously focused element. This element will be re-focused
     // when the dialog closes.
-    this._savePreviouslyFocusedElement();
-    // Move focus onto the dialog immediately in order to prevent the user
-    // from accidentally opening multiple dialogs at the same time.
-    this._focusDialogContainer();
+    if (this._document) {
+      this._elementFocusedBeforeDialogWasOpened = _getFocusedElementPierceShadowDom();
+    }
+
+    // Clicking on the backdrop will move focus out of dialog.
+    // Recapture it if closing via the backdrop is disabled.
+    this._overlayRef.backdropClick().subscribe(() => {
```

Comment:
If anything, I'd expose both backdrop and outside click since filtering out the backdrop clicks yourself can be tricky for folks that haven't had to do it.